modules/planning/tasks/optimizers/piecewise_jerk_speed/piecewise_jerk_speed_optimizer.cc

```diff
@@ -0,0 +1,157 @@
+    PathDecision* const path_decision, SpeedData* const speed_data) {
+  if (reference_line_info_->ReachedDestination()) {
+    return Status::OK();
+  }
+
+  if (path_data.discretized_path().empty()) {
+    std::string msg("Empty path data");
+    AERROR << msg;
+    return Status(ErrorCode::PLANNING_ERROR, msg);
+  }
+
+  for (const auto* obstacle : path_decision->obstacles().Items()) {
+    DCHECK(obstacle->HasLongitudinalDecision());
+  }
+  StGraphData& st_graph_data = *reference_line_info_->mutable_st_graph_data();
+
+  const auto& veh_param =
+      common::VehicleConfigHelper::GetConfig().vehicle_param();
+
+  std::array<double, 3> init_s = {0.0, st_graph_data.init_point().v(),
+                                       st_graph_data.init_point().a()};
+  double delta_t = 0.1;
+  std::array<double, 5> w = {1.0, 100.0, 10.0, 30.0, 0.0};
+  double total_length = st_graph_data.path_length_by_conf();
+  double total_time = st_graph_data.total_time_by_conf();
+  int num_of_knots = static_cast<int>(total_time / delta_t) + 1;
+  // Start a PathTimeQpProblem
+  std::unique_ptr<PathTimeQpProblem> path_time_qp(new PathTimeQpProblem());
+  path_time_qp->InitProblem(num_of_knots, delta_t, w, init_s);
+
+  path_time_qp->SetZeroOrderBounds(0.0, total_length);
+  path_time_qp->SetFirstOrderBounds(0.0, FLAGS_planning_upper_speed_limit);
+  path_time_qp->SetSecondOrderBounds(veh_param.max_deceleration(),
+                                     veh_param.max_acceleration());
+  path_time_qp->SetThirdOrderBound(FLAGS_longitudinal_jerk_bound);
+  // TODO(Hongyi): tune the params and move to a config
+  path_time_qp->SetFirstOrderBounds(0.0, 10.0);
+  path_time_qp->SetSecondOrderBounds(-4.4, 2.0);
+  path_time_qp->SetDesireDerivative(10.0);  // default speed to be 10.0
+
+  // Update STBoundary
+  std::vector<std::tuple<double, double, double>> x_bounds;
+  for (int i = 0; i < num_of_knots; ++i) {
+    double curr_t = i * delta_t;
+    double s_lower_bound = 0.0;
+    double s_upper_bound = total_length;
+    for (const STBoundary* boundary : st_graph_data.st_boundaries()) {
+      double s_lower = 0.0;
+      double s_upper = 0.0;
+      if (!boundary->GetUnblockSRange(curr_t, &s_upper, &s_lower)) {
+        continue;
+      }
+      switch (boundary->boundary_type()) {
+        case STBoundary::BoundaryType::STOP:
+        case STBoundary::BoundaryType::YIELD:
+          s_upper_bound = std::fmin(s_upper_bound, s_upper);
+          break;
+        case STBoundary::BoundaryType::FOLLOW:
+          // TODO(Hongyi): unify follow buffer on decision side
+          s_upper_bound = std::fmin(s_upper_bound, s_upper - 8.0);
+          break;
+        case STBoundary::BoundaryType::OVERTAKE:
+          s_lower_bound = std::fmax(s_lower_bound, s_lower);
+          break;
+        default:
+          break;
+      }
+    }
+    x_bounds.emplace_back(
+        std::make_tuple(curr_t, s_lower_bound, std::fmax(s_upper_bound, 0.0)));
```

Comment:
You may not need the `std::make_tuple` when using `emplace_back`. Please have a try.